marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flows/mapping-prov-test.sjs

```diff
@@ -0,0 +1,30 @@
+const test = require("/test/test-helper.xqy");
+const DataHubSingleton = require("/data-hub/5/datahub-singleton.sjs");
+const datahub = DataHubSingleton.instance();
+
+function describe(item) {
+  return xdmp.describe(item, emptySequence, emptySequence);
+}
+
+function fineProvOnMapping() {
+  datahub.flow.runFlow('CustomerMapping', 'prov-test-job', [
+    {
+      uri: '/customer1.json',
+      value: cts.doc('/customer1.json'),
+      context: {}
+    }
+  ], { provenanceGranularityLevel: 'fine'}, 1);
+  let provCount = fn.head(
+    xdmp.invokeFunction(function() {
+      return cts.estimate(cts.collectionQuery([
+        'http://marklogic.com/provenance-services/record'
+      ]));
+    }, { database: xdmp.database('data-hub-JOBS')})
+  );
+  return [
+    test.assertTrue(provCount > 1, `Provenance document count should be greater than 1 (was: ${provCount})`)
```

Comment:
I think it'd be useful to add assertions on one of the property-change provenance documents, just to verify and document that the expected content is in one of them. 

I noticed that a few of them have an encoded path as a type - e.g. "_2F__2A__3A_envelope_2F__2A__3A_instance_2F__2A__3A_postal". Is that expected?

Reply:
It needs to be encoded because the PROV library stores the value as an xs:QName. It may be encoded in more places that it really needs to. I'll look at isolating the encoded value to only the location that is storing the as an xs:QName.